platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -0,0 +1,247 @@
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Singleton;
+import org.jboss.logging.Logger;
+
+@Singleton
+public class K8Deployment {
+  private static final Logger LOGGER = Logger.getLogger(K8Deployment.class);
+  private final KubernetesClient client;
+  private final K8ConfigMap k8ConfigMap;
+  private final K8NameSpace k8NameSpace;
+
+  public K8Deployment(KubernetesClient client) {
+    this.client = client;
+    this.k8NameSpace = new K8NameSpace(client);
+    this.k8ConfigMap = new K8ConfigMap(client);
+  }
+
+  public String create(
+      DatacaterDeployment dcDeployment,
+      PipelineEntity pe,
+      StreamEntity streamIn,
+      StreamEntity streamOut) {
+    final String name = getName(dcDeployment.name());
+    final String volumeName = name + StaticConfig.VOLUME_NAME_SUFFIX;
+    k8NameSpace.create();
+
+    if (!pe.equals(new PipelineEntity())) {
+      k8ConfigMap.getOrCreate(name, pe);
+    }
+
+    if (exists(name)) {
+      return name;
+    }
+
+    List<EnvVar> variables = getEnvironmentVariables(dcDeployment, streamIn, streamOut);
+
+    var deployment =
+        new DeploymentBuilder()
+            .withNewMetadata()
+            .withName(name)
+            .addToLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endMetadata()
+            .withNewSpec()
+            .withReplicas(StaticConfig.REPLICAS)
+            .withMinReadySeconds(2)
+            .withNewSelector()
+            .addToMatchLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endSelector()
+            .withNewTemplate()
+            .withNewMetadata()
+            .addToLabels(
+                Map.of(
+                    StaticConfig.APP,
+                    StaticConfig.DATACATER_PIPELINE,
+                    StaticConfig.PIPELINE,
+                    StaticConfig.PIPELINE_NO,
+                    StaticConfig.REVISION,
+                    StaticConfig.PIPELINE_REV))
+            .endMetadata()
+            .withNewSpec()
+            .addNewContainer()
+            .withName(name)
+            .withImage(StaticConfig.EnvironmentVariables.FULL_IMAGE_NAME)
+            .withImagePullPolicy(StaticConfig.PULL_POLICY)
+            .withEnv(variables)
+            .withNewResources()
+            .withRequests(StaticConfig.RESOURCE_REQUESTS)
+            .withLimits(StaticConfig.RESOURCE_LIMITS)
```

Comment:
Can we make deployment configuration, like resource request/limits or number of replicas, configurable via the spec of the DataCater Deployment? This would add a lot of flexibility to deploying pipelines :)

Reply:
Sounds good. Could I open a ticket for this though? I think it could be a good first Ticket for our new Engineer, as it it touches on working with releavant subjects but doesn't require a deeper understanding of the tech behind it. And i guess it isn't a show stopper for our scope for the next three weeks

Reply:
great idea!

Reply:
created [a ticket](https://github.com/DataCater/datacater/issues/15) for this

Reply:
Thank you very much!